QModManager/PirateCheck.cs

```diff
@@ -162,28 +175,74 @@ private IEnumerator PlayVideo()
                     Process.Start("https://www.epicgames.com/store/en-US/product/subnautica-below-zero/home");
                     Process.Start("https://discordapp.com/store/skus/535869836748783616/subnautica-below-zero");
                 }
-                yield return new WaitForSeconds(35);
-                Process.Start("https://www.youtube.com/watch?v=dQw4w9WgXcQ");
-                
+                //yield return new WaitForSeconds(35);
+                //Process.Start("https://www.youtube.com/watch?v=dQw4w9WgXcQ");
 
                 while (videoPlayer.isPlaying)
                 {
                     yield return null;
                 }
 
                 yield return StartCoroutine(PlayVideo());
-            }*/
+            }
+
+            private static bool CheckConnection(string hostedURL = "http://www.google.com")
+            {
+                try
+                {
+                    string HtmlText = GetHtmlFromUri(hostedURL);
+                    if (HtmlText == "")
+                        return false;
+                    else
+                        return true;
+                }
+                catch (Exception)
+                {
+                    return false;
+                }
+            }
+            private static string GetHtmlFromUri(string resource)
+            {
+                string html = string.Empty;
+                HttpWebRequest req = (HttpWebRequest)WebRequest.Create(resource);
+                try
+                {
+                    using (HttpWebResponse resp = (HttpWebResponse)req.GetResponse())
+                    {
+                        bool isSuccess = (int)resp.StatusCode < 299 && (int)resp.StatusCode >= 200;
```

Comment:
Don't cast to `int`.  
Instead use the proper `HttpStatusCode` enum value
https://docs.microsoft.com/en-us/dotnet/api/system.net.httpstatuscode?view=netframework-4.7.2

Reply:
Not casting to an int would mean having to manually check for 5 to 7 values that can all represent success.

Reply:
You can still use `<` and `>=` with int based enums.

Reply:
Unless i'm missing something, the operators can only be used on the enums if they are casted to an integer beforehand

Reply:
It's probably a newer C# 7 feature. But it's perfectly valid and keeps arbitrary codes more readable

Reply:
Addressed in commit 5e0fb15